src/libsync/syncengine.cpp

```diff
@@ -1650,8 +1662,13 @@ bool SyncEngine::shouldDiscoverLocally(const QByteArray &path) const
         return true;
 
     auto it = _localDiscoveryPaths.lower_bound(path);
-    if (it == _localDiscoveryPaths.end() || !it->startsWith(path))
+    if (it == _localDiscoveryPaths.end() || !it->startsWith(path)) {
+        if (it != _localDiscoveryPaths.begin() && path.startsWith(*(--it))) {
```

Comment:
And this probably fails for similar reasons: if "foo" and "foo bar" are in the set, checking "foo/bar" would have <end> as lower bound, and the preceding entry isn't a prefix of the checked path. Tests for the space-case are needed.